itests/kit/node_unmanaged.go

```diff
@@ -464,332 +697,427 @@ func (tm *TestUnmanagedMiner) SubmitProveCommit(
+	minerInfo, err := tm.FullNode.StateMinerInfo(tm.ctx, tm.ActorAddr, head.Key())
 	if err != nil {
-		return fmt.Errorf("Miner(%s): failed to get miner info for sector %d: %w", tm.ActorAddr, sectorNumber, err)
+		return fmt.Errorf("Miner(%s): failed to get miner info: %w", tm.ActorAddr, err)
 	}
 
-	r, err := tm.SubmitMessage(ctx, &miner14.SubmitWindowedPoStParams{
-		ChainCommitEpoch: chainRandomnessEpoch,
-		ChainCommitRand:  chainRandomness,
-		Deadline:         sp.Deadline,
-		Partitions:       []miner14.PoStPartition{{Index: sp.Partition}},
-		Proofs:           []proof.PoStProof{{PoStProof: minerInfo.WindowPoStProofType, ProofBytes: proofBytes}},
-	}, 0, builtin.MethodsMiner.SubmitWindowedPoSt)
-	if err != nil {
-		return fmt.Errorf("Miner(%s): failed to submit window post for sector %d: %w", tm.ActorAddr, sectorNumber, err)
+	postMessages := make([]cid.Cid, 0)
+
+	submit := func(partitions []uint64, sectors []sectorInfo) error {
+		var proofBytes []byte
+		if tm.mockProofs {
+			proofBytes = []byte{0xde, 0xad, 0xbe, 0xef}
+		} else {
+			proofBytes, err = tm.generateWindowPost(sectors)
+			if err != nil {
+				return fmt.Errorf("Miner(%s): failed to generate window post for deadline %d, partitions %v: %w", tm.ActorAddr, di.Index, partitions, err)
+			}
+		}
+
+		tm.log("WindowPoST submitting %d sectors for deadline %d, partitions %v", len(sectors), di.Index, partitions)
+
+		pp := make([]miner14.PoStPartition, len(partitions))
+		for i, p := range partitions {
+			pp[i] = miner14.PoStPartition{Index: p}
+		}
+
+		// Push all our post messages to the mpool and wait for them later. The blockminer might have
+		// ceased mining while it waits to see our posts in the mpool and if we wait for the first
+		// message to land but it's waiting for a later sector then we'll deadlock.
+		mCid, err := tm.mpoolPushMessage(&miner14.SubmitWindowedPoStParams{
+			ChainCommitEpoch: chainRandomnessEpoch,
+			ChainCommitRand:  chainRandomness,
+			Deadline:         di.Index,
+			Partitions:       pp,
+			Proofs:           []proof.PoStProof{{PoStProof: minerInfo.WindowPoStProofType, ProofBytes: proofBytes}}, // can only have 1
+		}, 0, builtin.MethodsMiner.SubmitWindowedPoSt)
+		if err != nil {
+			return fmt.Errorf("Miner(%s): failed to submit PoSt for deadline %d, partitions %v: %w", tm.ActorAddr, di.Index, partitions, err)
+		}
+
+		postMessages = append(postMessages, mCid)
+
+		return nil
 	}
 
-	if !r.Receipt.ExitCode.IsSuccess() {
-		return fmt.Errorf("Miner(%s): submitting PoSt for sector %d failed: %s", tm.ActorAddr, sectorNumber, r.Receipt.ExitCode)
+	toSubmitPartitions := make([]uint64, 0)
+	toSubmitSectors := make([]sectorInfo, 0)
+	for partition, sectors := range partitionMap {
+		if len(sectors) == 0 {
+			continue
+		}
+		toSubmitPartitions = append(toSubmitPartitions, partition)
+		toSubmitSectors = append(toSubmitSectors, sectors...)
+		if len(toSubmitPartitions) == miner14.PoStedPartitionsMax {
```

Comment:
So the limit is on the number of partitions we can prove in a single Post but not on the number of sectors that we can prove per partition ?

Reply:
Correct, it's fixed to `3` but I'm using the constant for it anyway. The limit for the number of sectors that can be proven is on the number of sectors per partition, and that depends on the sector size: https://github.com/filecoin-project/ref-fvm/blob/844d34a2228a9de7f86c3ca86f9efbbf9ea4215b/shared/src/sector/registered_proof.rs#L175-L188

It's 2 sectors per partition for 2k sectors so that's 6 sectors that can be proven in one post max for 2k.